app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -3,8 +3,51 @@ kind: registration
 name: Global Registration
 model: ProvisioningTemplate
 -%>
+<%
+    curl_opts = ["-H 'Content-Type: application/json'", "-H 'Accept: application/json'", "-H 'Authorization: Bearer #{@auth_token}'"]
+%>
+#!/bin/bash
+
+# Template parameters:
+<%= "#  - User: [#{@user.login}]" if @user %>
+<%= "#  - Organization: [#{@organization.name}]" if @organization %>
+<%= "#  - Location: [#{@location.name}]" if @location %>
+<%= "#  - Hostgroup: [#{@hostgroup.name}]" if @hostgroup %>
 
 if ! [ $(id -u) = 0 ]; then
     echo "Please run as root"
     exit 1
 fi
+
+if [ -f /etc/os-release ] ; then
+  . /etc/os-release
+fi
+
+HOSTNAME=$(hostname -f)
+
+echo "#"
+echo "# Running registration"
+echo "# Hostname: $HOSTNAME"
+echo "# OS: $NAME"
+echo "#"
+
+if [ $ID = rhel ] ; then
+    echo "RHEL registration"
+<% if plugin_present?('katello') %>
+    CONSUMER_RPM=$(mktemp).rpm
+
+    curl --insecure --output $CONSUMER_RPM <%= foreman_server_url %>/pub/katello-ca-consumer-latest.noarch.rpm
+    yum localinstall $CONSUMER_RPM -y
+    yum install subscription-manager -y
+    subscription-manager register --org=<%= @organization.label %> --activationkey=key1
+    rm -f $CONSUMER
+<% end %>
+else
+    curl -X POST <%= foreman_server_url %>/api/hosts \
+     <%= curl_opts.join(' ') %> \
+     -d "{\"host\": {\"name\": \"$HOSTNAME\", \"build\": \"false\", \"managed\": \"false\" \
```

Comment:
It may be easier to use `'{"host": {"name": "'$HOSTNAME'", ` so you don't need to escape so much.

Reply:
Or pipe a heredoc into curl